From: Mykola Golub <mgolub@mirantis.com>
Date: Wed, 4 May 2016 15:23:11 +0300
Subject: rbd-mirror: don't unregister asok commands if image replayer start
 failed

Signed-off-by: Mykola Golub <mgolub@mirantis.com>
(cherry picked from commit 2fd6fdd3dc9ecb915362f5e3aa8a224210273540)
(cherry picked from commit 66f7f2d7bda7f2db6871870f5622594d6391b921)
---
 src/tools/rbd_mirror/ImageReplayer.cc | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/tools/rbd_mirror/ImageReplayer.cc b/src/tools/rbd_mirror/ImageReplayer.cc
index 203b78c..9ec48e2 100644
--- a/src/tools/rbd_mirror/ImageReplayer.cc
+++ b/src/tools/rbd_mirror/ImageReplayer.cc
@@ -562,9 +562,6 @@ void ImageReplayer<I>::on_start_fail_finish(int r)
   m_local_ioctx.close();
   m_remote_ioctx.close();
 
-  delete m_asok_hook;
-  m_asok_hook = nullptr;
-
   if (on_start_finish != nullptr) {
     dout(20) << "on start finish complete, r=" << r << dendl;
     on_start_finish->complete(r);
